### DISCUSSION TITLE
Force Calculation

### DISCUSSION THREAD

**Discussion by dineshkumarreddyg36** [xcompact3d/Incompact3d#111] [Q&A] [answered]
I've done the forced oscillation of the cylinder by modifying the **BC-Cylinder.f90** file and **ibm.f90** file.  The changes that I've made are shown below and it worked successfully. 
**ceyy=cey+0.5*sin(2*3.141592654*ubcy*t)
cexx=cex**

The **control volume** for the **forces** is taken as follows

**"&ForceCVs

xld(1) = 3.0 !X left for volume control
xrd(1) = 7.0 !X right for volume control
yld(1) = 2.0 !Y bottom for volume control
yud(1) = 4.0 !Y top for volume control
!zld(1) = 0.01
!zrd(1) = 9.99

/End"**

And the **Coefficient of lift** is plotted with time and is as shown
![C_l](https://user-images.githubusercontent.com/106377425/183076572-8c2829d2-6b7d-481a-b5fe-c10273eee5d7.PNG)

The lift coefficient seems not reasonable.
Am I making any mistake in taking the control volume around the cylinder for force calculation? Can someone comment on it?
Also I'm trying to do the VIV(Vortex Induced Vibration) on the Cylinder where I'm actually facing the same problem. 

The animation of the cylinder oscillation is shown below

https://user-images.githubusercontent.com/106377425/183083465-262ad33f-59c7-46ce-883a-88a27974a588.mp4

Thanks in advance..

**Comment by nasos94** [comment, marked as answer]
The oscillations which occur are to be expected (they are commonly known as spurious force oscillations (SFOs)). The oscillations will decrease in amplitude with the increase of the spatial resolution and the increase of the time step. The simplest solution is to filter the hydrodynamic coefficients in a post-processing step which should work just fine for both cases mentioned.

With a simple filtering you should be able to get accurate predictions for the hydrodynamic coefficients. You can take a look at our recent paper where we have performed simulations of the VIVs of a cylinder at various Reynolds numbers here: https://trebuchet.public.springernature.app/get_content/f00e5a2b-c2b8-437e-9bfd-078dd012ff44 

For more information on the filtering you can also look at the following paper: https://www.sciencedirect.com/science/article/pii/S0307904X21003103

From the sort video it looks like the domain may be too small in the Y-direction and perhaps too wide in the X-direction. A final comment, please ensure that the surrounding box which is used for the calculation of the hydrodynamic forces is large enough to capture the amplitude of the cylinder oscillations (the coordinates of the box are printed by the **forces.f90** script during the first iterations of the solver.)

Hope this helps!

**Reply by dineshkumarreddyg36** [reply]
Got it, Thank You @nasos94
